memory.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+
+class Memory():
+
+    def __init__(self, max_size, observation_size):
+        """
+        max_size: int
+            The max amount of samples we want to hold until we start 
+            deleting the oldest sample
+
+        observation_size: tuple
+            A tuple describing a single observations size from the environment. 
+            Does NOT include the batch size.
+            This module will try to be agnostic to the size pass as long as its 
+            an image.
+            eg. (84, 84, 3) or (3, 84, 84) or (84,84)
```

Comment:
we can keep it open, it shouldn't matter as long as the tuple for the format is passed in! When we are doing things like resizing the image we could assume the smallest value (if there are more than 2) is the channel.